gh-pages/_posts/how-to/2020-08-17-fixate-folders-with-a-custom-cc-json.md

```diff
@@ -0,0 +1,98 @@
+---
+categories:
+    - How-to
+title: Fixate folders using a custom cc.json
+---
+
+# Introduction
+
+The [Squarified-Tree-Map-Algorithm](https://www.win.tue.nl/~vanwijk/stm.pdf) generates a layout for the visualized map.
+The layout varies depending on the area of each node. Adding a new file to a folder might increase the area of the parent folder,
+so that the algorithm decides that the parent-folder needs to be relocated. Another way to follow this problem is by changing the area-metric.
+Notice, that most folders will not be located at the same position as before.
+
+# Fixate folders
+
+In order to prevent folders from changing locations, we introduced a new attribute to the `.cc.json` called `fixed`.
+This attribute will not be generated during the analysis. The user needs to edit the `.cc.json` by hand.
+
+# API
+
+```json
+ "fixed": {
+    "x": 60,
+    "y:": 40,
+    "width": 35,
+    "height": 55
+ }
+```
+
+The `fixed`-attribute includes a coordinate `x` and `y`. These represent the top-left corner of the folder. `width`
+defines the length in x-direction and `height` defines the length in y-direction.
+
+Note, that these numbers represent percentages. The overall map size is therefore `100`.
+
+## Example
+
+```json
+{
+	"projectName": "example-project",
+	"apiVersion": "1.2",
+	"nodes": [
+		{
+			"name": "root",
+			"type": "Folder",
+			"attributes": {},
+			"children": [
+				{
+					"name": "folder_1",
+					"type": "Folder",
+					"attributes": {},
+					"children": [
+						{
+							"name": "children_1",
+							"type": "File",
+							"attributes": {
+								"custom_metric": 2
+							},
+							"children": []
+						}
+					],
+					"fixed": {
+						"x": 5,
+						"y:": 5,
+						"width": 50,
+						"height": 30
+					}
+				},
+				{
+					"name": "folder_2",
+					"type": "Folder",
+					"attributes": {},
+					"children": [],
+					"fixed": {
+						"x": 60,
+						"y:": 40,
+						"width": 35,
+						"height": 55
+					}
+				}
+			]
+		}
+	]
+}
+```
+
+Notice, that `folder_1` start at `x:5`, `y:5`, because we apply a margin of `5` around the map.
+`folder_2` starts at `x:60`, `y:40`, because `folder_1` ends at `x:55`, `y:35` and we apply another margin of `5` between those.
+
+## Restrictions
+
+![Example]({{site.baseurl}}/assets/images/posts/how-to/fixate-folders/fixate-folder-example.jpg)
+
+In order to build a valid custom `.cc.json`, these rules must be followed:
+
+-   The value `x + width` or `y + height` must be in a range of `[0, 100]`.
+-   Folders can't overlap. Applying a margin between folders is recommended.
```

Comment:
The margin can not be changed by the user, right? If that's the case, the second sentence seems to be redundant?

Reply:
The margin is determined by the space that is left over after you defined your `fixed`-object. The margin can be different on every side.

Maybe `Leaving some space between folders for visibility reasons is recomended.` is better?

Reply:
Smart. Even though we can't determine all bad input cases due to that. Seems like it's a well made consideration.

Reply:
Your suggestion sound pretty good. I'd go with

```suggestion
-   Folders may not overlap.
-   Leaving space between folders (for visibility reasons) is recomended.
```